staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/validator.go

```diff
@@ -65,48 +65,50 @@ func (a customResourceValidator) Validate(ctx context.Context, obj runtime.Objec
 	}
 	customResource := customResourceObject.UnstructuredContent()
 
+	allErrs := validation.ValidateObjectMetaAccessor(accessor, a.namespaceScoped, validation.NameIsDNSSubdomain, field.NewPath("metadata"))
```

Comment:
also, wanted to confirm the rational behind not bundling up the `kind` and `apiVersion` errors and returning early in those cases: is it because any error in those would mean a completely wrong "type" of object and so we return early?

Reply:
yes. They are more critical.